lib/Mojolicious.pm

```diff
@@ -0,0 +1,1217 @@
+command object and the command arguments)
+
+=head2 before_server_start
+
+Emitted right before the application server is started, for web servers that support it, which includes all the
+built-in ones.
+
+  $app->hook(before_server_start => sub ($server, $app) {...});
+
+Useful for reconfiguring application servers dynamically or collecting server diagnostics information. (Passed the
+server and application objects)
+
+=head2 after_build_tx
+
+Emitted right after the transaction is built and before the HTTP request gets parsed.
+
+  $app->hook(after_build_tx => sub ($tx, $app) {...});
+
+This is a very powerful hook and should not be used lightly, it makes some rather advanced features such as upload
+progress bars possible. Note that this hook will not work for embedded applications, because only the host application
+gets to build transactions. (Passed the transaction and application objects)
+
+=head2 around_dispatch
+
+Emitted right after a new request has been received and wraps around the whole dispatch process, so you have to
+manually forward to the next hook if you want to continue the chain. Default exception handling with
+L<Mojolicious::Plugin::DefaultHelpers/"reply-E<gt>exception"> is the first hook in the chain and a call to
+L</"dispatch"> the last, yours will be in between.
+
+  $app->hook(around_dispatch => sub ($next, $c) {
+    ...
+    $next->();
+    ...
+  });
+
+This is a very powerful hook and should not be used lightly, it allows you to, for example, customize application-wide
+exception handling, consider it the sledgehammer in your toolbox. (Passed a callback leading to the next hook and the
+default controller object)
+
+=head2 before_dispatch
+
+Emitted right before the static file server and router start their work.
+
+  $app->hook(before_dispatch => sub ($c) {...});
+
+Very useful for rewriting incoming requests and other preprocessing tasks. (Passed the default controller object)
+
+=head2 after_static
+
+Emitted after a static file response has been generated by the static file server.
+
+  $app->hook(after_static => sub ($c) {...});
+
+Mostly used for post-processing static file responses. (Passed the default controller object)
+
+=head2 before_routes
+
+Emitted after the static file server determined if a static file should be served and before the router starts its
+work.
+
+  $app->hook(before_routes => sub ($c) {...});
+
+Mostly used for custom dispatchers and collecting metrics. (Passed the default controller object)
+
+=head2 around_action
+
+Emitted right before an action gets executed and wraps around it, so you have to manually forward to the next hook if
+you want to continue the chain. Default action dispatching is the last hook in the chain, yours will run before it.
+
+  $app->hook(around_action => sub ($next, $c, $action, $last) {
+    ...
+    return $next->();
+  });
+
+This is a very powerful hook and should not be used lightly, it allows you for example to pass additional arguments to
```

Comment:
Perhaps in the cookbook document an example of passing additional arguments to actions and an example of handling return values differently.  From a centralized hooks cookbook section, if there are other places where it's already documented, it can just be pointed to...